content/basics/basics.mdx

```diff
@@ -0,0 +1,327 @@
+
+## When do observers observe?
+
+Note that MobX can only observe properties, not simple values like a `string`
+or a `number`. So if you write your component to only receive simple values, it
+does not benefit from observability:
+
+```tsx
+import { Component } from 'react'
+import { observer } from 'mobx-react'
+
+// Uses observer() but does not benefit from observability
+const NonObservingAnimalView = observer(
+  ({
+    name,
+    species,
+    isHungry,
+  }: {
+    name: string
+    species: string
+    isHungry: boolean
+  }) => {
+    return (
+      <div>
+        {name} is of species {species} and is{' '}
+        {isHungry ? 'hungry' : 'not hungry'}.
+      </div>
+    )
+  },
+)
+```
+
+The component `NonObservingAnimalView` works perfectly if it is used within
+another component that receives an `Animal` instance and is also an observer,
+but it re-render each and every time, because MobX cannot see whether
+`name`, `species` or `isHungry` are changed. Our original `AnimalView` _does_
+observe it, because it can track property accesses.
+
+Wrapping a component in `observer` is cheap, so if you use MobX with react,
+it's recommended to just wrap all your components in `observer`, even if the
+component does not use any observable information yet. There is no need
+distinguish between React components that observe and do not observe in this.
+
+## Observable arrays
+
+MobX does support observable arrays, observable
+objects and observable maps, so those _can_ be observed for changes by
+a React component. We demonstrate that here with `animals`:
+
+```tsx
+import { Component } from 'react'
+import { observer } from 'mobx-react'
+
+const AnimalsView = observer(({ animals }: { animals: Animal[] }) => {
+  return (
+    <ul>
+      {animals.map(animal => (
+        <ul>
+          <AnimalView key={animal.name} animal={animal} />
+        </ul>
+      ))}
+    </ul>
+  )
+})
+```
+
+Instances of our `Zoo` class have an `animals` property that is observable, so
+let's use that:
+
+```tsx
+const wilma = new Animal('Wilma', 'penguin')
+
+const zoo = new Zoo()
+zoo.addAnimal(fred)
+zoo.addAnimal(wilma)
+```
+
+We can now render our animals:
+
+```tsx
+<AnimalsView animals={zoo.animals} />
+```
+
+`AnimalsView` automatically re-renders itself whenever you change the array, in
+this case by using `Zoo.addAnimal`. The individual `AnimalView` re-renders
+itself as before when any observed properties the underlying object change,
+which in this case is its `isHungry` status.
+
+## Local component state with `useLocalStore`
+
+So far we have discussed observable model state, where the models live outside
+of the React components. It is useful to have component-specific state however.
+This is state that only makes sense within that component, is typically very UI
+specific, and there is no reason to specify this part of your application's
+models.
+
+This is useful for "temporary data", such as for instance the text that a user
```

Comment:
I don't like the way you are treating `useLocalStore` here. It's far from "temporary data" only. It's a regular observable you can pass around, including actions & computeds. You can even stick into the Context and use anywhere in the app. It's equivalent to class-based stores. I know, the name of that hook tells something else, but it's too late to change that now.

Reply:
This is a tricky issue. You're right of course in that you can do anything with useLocalState. But I want to make sure the documentation doesn't offer too many options at once, and highlights the model-driven nature of many MobX applications. And I do want to mention the use case of React component specific state. I've added a bit of language to make clear it can do more for you too.

Reply:
I agree. As I said in a comment below, we shouldn't mention `useLocalStore` here at all. Feel free to extract this portion to a separate file that won't be linked anywhere and we will see how it ends up.

Reply:
This means that a very commonly used pattern with MobX has no equivalent with functional components. I mean this one:

```js
class MyReactComponent extends Component {
  @observable isOpen = false
}
```
 
Are functional components with hooks read for prime time?